cosmo/monitors/nuv_dark_monitor.py

```diff
@@ -28,24 +28,29 @@ def get_data(self):
         # access data, perform any filtering required for analysis
         data = self.model.new_data
         dark_rate_column = []
+        dec_year_column = []
 
         xlim = [0, 1024]
         ylim = [0, 1024]
 
         # parallelize, this is going to get bad when looking at a lot of data
         for index, row in data.iterrows():
             subdf = pd.DataFrame({
-                "TIME": row["TIME"], "XCORR": row["XCORR"],
-                "YCORR": row["YCORR"]
+                "EXPSTART": row["EXPSTART"], "TIME": row["TIME"],
+                "XCORR": row["XCORR"], "YCORR": row["YCORR"],
+                "TIME_3": row["TIME_3"]
                 })
             filtered = subdf.where(
                 (subdf["XCORR"] > xlim[0]) & (subdf["XCORR"] < xlim[1]) & (
                             subdf["YCORR"] > ylim[0]) & (
                             subdf["YCORR"] < ylim[1]))
```

Comment:
You could do it this way, but you can't add the array elements directly here. You'd need to put them in another list so that the number of items for each col matches: 

```python
for idx, row in data.iterrows():
    subdf = pd.DataFrame(
        {
            'EXPSTART': row['EXPSTART'],
            'TIME': [row['TIME']],
            'XCORR': [row['XCORR']],
            ...
        }
    )
    ...
```

However, you might be able to use the `apply` method here instead:
```python
data = data.apply(lambda row: row[np.where((row.XCORR > xlim[0]) & (row.XCORR < xlim[1]) & ...)]. axis=1)
```

However, you'd then need to get rid of rows with empty arrays in the "array columns" or deal with that later. You can find an example of this here: 
https://github.com/spacetelescope/cosmo/blob/5ad7ccbe099b0977d3e8f32cac10a3954af68cd4/cosmo/monitors/data_models.py#L125

Reply:
thanks i am dumb